pep/app/pods/components/page/sidebar/widgets/sign-in/component.ts

```diff
@@ -0,0 +1,141 @@
+    @tracked loginError = null;
+    @tracked logins = null;
+    @tracked padsForgotPasswordUrl = null;
+    @tracked genericLoginUrl = null;
+    @tracked padsLoginUrl = null;
+    @tracked padsRegisterUrl = null;
+
+    /**
+     * Create the loginForm changeset.
+     * @param {unknown} owner
+     * @param {PageSidebarWidgetsSignInArgs} args
+     */
+    constructor(owner: unknown, args: PageSidebarWidgetsSignInArgs) {
+        super(owner, args);
+        const model: LoginForm = { username: null, password: null };
+        const changeset = createChangeset<LoginForm>(model, LoginValidations);
+        this.changeset = changeset;
+    }
+
+    get cardInfo() {
+        return this.configuration.content.global.signInCard;
+    }
+
+    /**
+     * Forgot password url. We use the FP URL from the api, and then the hardcoded one as a backup
+     *
+     * @readonly
+     */
+    get forgotPasswordUrl() {
+        return this.padsForgotPasswordUrl ?? FORGOT_PW_URL;
+    }
+
+    /**
+     * Submits the login form and logs the user in
+     * @param {GenericChangeset<LoginForm>} changeset
+     */
+    @action
+    async login(changeset: GenericChangeset<LoginForm>) {
+        try {
+            const username = changeset.username;
+            const password = changeset.password;
+            this.loadingBar.show();
+            const response = await this.session.authenticate('authenticator:credentials', username, password);
+            this.loginError = null;
+            this.loadingBar.hide();
+            this.notifications.success(this.intl.t('login.success'));
+            await onAuthenticated(this);
+            return response;
+        } catch (err) {
+            this.loginError = err;
+            this.loadingBar.hide();
+            return reject(err);
+        }
+    }
+
+    /**
+     * Load the widget login data
+     */
+    @restartableTask
+    *loadResults() {
+        const session = this.session.isAuthenticated
+            ? this.session.data?.authenticated?.SessionId
+            : this.session.getUnauthenticatedSession()?.SessionId;
+        const params = serializeQueryParams({
+            sessionId: session ?? ''
+        });
+        const federatedLogins = yield this.ajax.request<FederatedLoginResponse>(`${ENV.federatedLoginUrl}?${params}`, {
+            appendTrailingSlash: false
+        });
+        this.logins = JSON.parse(federatedLogins.FederatedLinks).FederatedLinks;
+        this.genericLoginUrl = federatedLogins.GenericFederatedURL;
+        this.padsLoginUrl = federatedLogins.PaDSLogonURL;
+        this.padsForgotPasswordUrl = federatedLogins.PaDSPasswordResetURL;
+        this.padsRegisterUrl = federatedLogins.PaDSRegisterUserURL;
+    }
+
+    /**
+     * Load the widget results on render
+     *
+     */
+    @action
+    @dontRunInFastboot
+    onElementInsert() {
+        later(() => {
+            taskFor(this.loadResults).perform();
+        }, 1000);
```

Comment:
shouldnt have this delay for this card, I don't think. these delays were just a workaround for some other weird race condition with the main PEP API.